pkg/operator/staticpod/controller/startupmonitorcondition/startup_monitor_condition.go

```diff
@@ -0,0 +1,150 @@
+package startupmonitorcondition
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	corev1listers "k8s.io/client-go/listers/core/v1"
+)
+
+// startupMonitorPodConditionController monitors the startup monitor pod and reports unwanted state/condition/phase as a degraded condition
+type startupMonitorPodConditionController struct {
+	operatorClient operatorv1helpers.OperatorClient
+
+	targetName string
+	podLister  corev1listers.PodNamespaceLister
+
+	startupMonitorEnabledFn func() bool
+}
+
+// New returns a controller for monitoring the lifecycle of the startup monitor pod
+func New(targetNamespace string,
+	targetName string,
+	operatorClient operatorv1helpers.OperatorClient,
+	kubeInformersForNamespaces operatorv1helpers.KubeInformersForNamespaces,
+	startupMonitorEnabledFn func() bool,
+	eventRecorder events.Recorder) factory.Controller {
+	fd := &startupMonitorPodConditionController{
+		operatorClient:          operatorClient,
+		targetName:              targetName,
+		podLister:               kubeInformersForNamespaces.InformersFor(targetNamespace).Core().V1().Pods().Lister().Pods(targetNamespace),
+		startupMonitorEnabledFn: startupMonitorEnabledFn,
+	}
+	return factory.New().WithSync(fd.sync).ResyncEvery(6*time.Minute).WithInformers(kubeInformersForNamespaces.InformersFor(targetNamespace).Core().V1().Pods().Informer()).ToController("StartupMonitorPodCondition", eventRecorder)
+}
+
+func (fd *startupMonitorPodConditionController) sync(_ context.Context, _ factory.SyncContext) (err error) {
+	startupPodPendingDegraded := &operatorv1.OperatorCondition{Type: "StartupMonitorPodPendingDegraded", Status: operatorv1.ConditionFalse}
+	startupPodContainerWaitingDegraded := &operatorv1.OperatorCondition{Type: "StartupMonitorPodContainerWaitingDegraded", Status: operatorv1.ConditionFalse}
+	startupPodFailedContainerDegraded := &operatorv1.OperatorCondition{Type: "StartupMonitorPodFailedContainerDegraded", Status: operatorv1.ConditionFalse}
+	startupPodContainerExcessiveRestartsDegraded := &operatorv1.OperatorCondition{Type: "StartupMonitorPodContainerExcessiveRestartsDegraded", Status: operatorv1.ConditionFalse}
```

Comment:
these are many conditions. I don't think we want one for each symptom, but rather use the reason and message to distinguish.